src/main/java/build/buildfarm/instance/memory/MemoryInstance.java

```diff
@@ -132,8 +132,14 @@ public InputStream newStreamInput(String name) {
 
   @Override
   public Iterable<Digest> findMissingBlobs(Iterable<Digest> digests) {
-    return Iterables.filter(
-        digests, digest -> digest.getSizeBytes() != 0 && !contentAddressableStorage.contains(digest));
+    ImmutableList.Builder<Digest> missingBlobs = new ImmutableList.Builder<>();
```

Comment:
Can all of getBlob, putBlob, and findMissingBlobs have the respective 0 case workaround in the AbstractServerInstance? IIUC, that will ensure that no matter which CAS implementation is used in the config, the invariant that 0-length blobs always virtually exist will be maintained. Basically, I'm just trying to insert this at a level such that the code is both guaranteed to exist and won't need to be duplicated.

Reply:
Further agreed, I am not 100% sure why I had the MemoryInstance hold the findMissingBlobs impl, but perhaps I just missed it my initial refactor.